src/Document/Obstacles/Option.ts

```diff
@@ -0,0 +1,192 @@
+import { bzwBool, bzwInt, bzwRepeatable, bzwString } from '../attributeParsers';
+import { IBaseObject, newIBaseObject } from './BaseObject';
+
+export type Accelerations = {
+  linear: number;
+  angular: number;
+};
+
+export type BZDBSetting = {
+  name: string;
+  value: string;
+};
+
+export type ShotLimit = {
+  id: string;
+  num: number;
+};
+
+export type FlagCount = {
+  id: string;
+  count: number;
+};
+
+export enum RabbitMode {
+  score = 'score',
+  killer = 'killer',
+  random = 'random',
+}
+
+export type MaxPlayers = {
+  total?: number;
+  rogue?: number;
+  red?: number;
+  green?: number;
+  blue?: number;
+  purple?: number;
+  observer?: number;
+};
+
+export function bzwAccelerations(input: string): Accelerations {
+  const accArray = input.split(' ');
+  const linear = accArray[0] ? parseInt(accArray[0]) : 50;
+  const angular = accArray[1] ? parseInt(accArray[1]) : 38;
+
+  return {
+    linear: linear,
+    angular: angular,
+  };
+}
+
+export function bzwBZDBSetting(input: string): BZDBSetting {
+  const DBArray = input.split(' ');
+  const name = DBArray[0];
+  const value = DBArray[1];
+
+  return {
+    name: name,
+    value: value,
+  };
+}
+export function bzwShotLimit(input: string): ShotLimit {
+  const slArray = input.split(' ');
+  const id = slArray[0];
+  const num = parseInt(slArray[1]);
+
+  return {
+    id: id,
+    num: num,
+  };
+}
+
+export function bzwRabbitMode(value: string): RabbitMode {
+  return value as RabbitMode;
+}
+
+export function bzwMaxPlayers(input: string): MaxPlayers {
+  const maxMPArray = input.split(',');
+  if (maxMPArray.length === 1) {
+    const total = parseInt(maxMPArray[0]);
+    return {
+      total: total,
+    };
+  }
+  return {
+    rogue:
+      maxMPArray[0] && maxMPArray[0] === ''
+        ? undefined
+        : parseInt(maxMPArray[0]),
```

Comment:
I'm trying to wrap my head around this condition. Isn't an empty string considered falsey? wouldn't this short circuit and always evaluate to false?

Reply:
Uright :/